src/main/kotlin/app/hashers/CodeLongevity.kt

```diff
@@ -107,7 +125,68 @@ class CodeLine(val repo: Repository,
         val state = if (isDeleted) "deleted" else "alive"
         return "Line '$text' - '${from.file}:${from.line}' added in $fc $fd\n" +
             "  ${revState} '${to.file}:${to.line}' in $tc $td,\n" +
-            "  age: ${age} ms - $state"
+            "  age: ${age} s - $state"
+    }
+
+    private var _age : Long = 0;
+}
+
+/**
+ * Detects colleagues and their 'work vicinity' from commits.
+ */
+class Colleagues {
+    private val map: HashMap<Pair<String, String>,
+                             HashMap<String, Long>> = hashMapOf()
+
+    fun collect(line: CodeLine) {
+        // TODO(alex): ignore same user emails
+        val authorEmail = line.authorEmail
+        val editorEmail = line.editorEmail
+        if (editorEmail == null || authorEmail == editorEmail) {
+            return
+        }
+        val emails =
+            if (editorEmail < authorEmail) Pair(editorEmail, authorEmail)
+            else Pair(authorEmail, editorEmail)
+
+        val dates = map.getOrPut(emails, { hashMapOf() })
+        val month = SimpleDateFormat("yyyy-MM").format(line.editDate)
+
+        println("collected colleague, age: ${line.age}")
+        var vicinity = dates.getOrPut(month, { line.age })
+        if (vicinity > line.age) {
+            dates.put(month, line.age)
+        }
+    }
+
+    fun updateStats() {
+        // TODO(alex): report the stats to the server
+        if (Logger.isDebug) {
+            map.forEach( { (email1, email2), dates ->
+                println("<$email1> <$email2>")
```

Comment:
Logger.debug

Reply:
ok, not big deal though as the code should go away